sql/core/src/main/scala/org/apache/spark/sql/execution/basicPhysicalOperators.scala

```diff
@@ -78,9 +78,42 @@ case class ProjectExec(projectList: Seq[NamedExpression], child: SparkPlan)
     }
   }
 
-  override def outputOrdering: Seq[SortOrder] = child.outputOrdering
```

Comment:
Yeah, I see. In order to support general cases, I am going first improve expression canonicalization. So we can check semantic equal between `(a + b) - b` and `a`.